fix: bump temperature to 16 bits

jhonasiv opened the pull request:
Depends on:
- [ ] https://github.com/rock-drivers/drivers-canopen_master/pull/8

Latest firmware change these parameters to be 16 bit. We need to do some work to allow for higher sized types when parsing canopen messages.

jhonasiv committed:
fix: bump temperature to 16 bits

Latest firmware change these parameters to be 16 bit. We need to do some
work to allow for higher sized types when parsing canopen messages.

doudou reviewed (changes requested):
BLOCKER: we can't merge this until we have changed canopen_master to allow the C++ type to be bigger than the canopen object, or we will break all other usages of the driver.

doudou committed:
fix: status via PDO in tests

With firmwares that have 2 bytes per message, we are restricted
to two channels to use that feature.
